src/updateIndex.ts

```diff
@@ -0,0 +1,82 @@
+import * as fs from "fs";
+import * as path from "path";
+import { ProgressLocation, window } from "vscode";
+import { executeJavaLanguageServerCommand } from "./jdtls/commands";
+const axios = require("axios");
+
+// tslint:disable-next-line: export-name
+export async function updateIndex(extensionPath: string): Promise<void> {
+    axios.defaults.timeout = 6000;
+    window.withProgress({
+        location: ProgressLocation.Notification,
+        title: "Updating artifacts index data...",
+        cancellable: true
+    }, async (_progress, _token) => {
+        return doUpdateIndex(extensionPath);
+    });
+}
+
+async function doUpdateIndex(extensionPath: string): Promise<void> {
+    const baseDir: string = extensionPath;
+    const indexResourceUrl: string = "https://api.github.com/repos/RunxinXu/IndexInfo/contents/index";
```

Comment:
personal url?

Reply:
ohhh, I am not sure if the index data can be reached when the PR is not merged...  I will modify it ...